src/dispatch/workflow/service.py

```diff
@@ -83,22 +86,48 @@ def get_instance(*, db_session, instance_id: int) -> WorkflowInstance:
     )
 
 
-def create_instance(*, db_session, instance_in: WorkflowInstanceCreate) -> WorkflowInstance:
+def get_running_instances(*, db_session) -> List[WorkflowInstance]:
+    """Fetches all running instances."""
+    return (
+        db_session.query(WorkflowInstance)
+        .filter(
+            WorkflowInstance.status.in_(
+                (
+                    WorkflowInstanceStatus.created,
+                    WorkflowInstanceStatus.running,
+                    WorkflowInstanceStatus.submitted,
+                )
+            )
+        )
+        .all()
+    )
+
+
+def create_instance(
+    *, db_session, workflow, instance_in: WorkflowInstanceCreate
+) -> WorkflowInstance:
     """Creates a new workflow instance."""
     instance = WorkflowInstance(
-        **instance_in.dict(exclude={"incident", "workflow", "creator", "artifacts"})
+        **instance_in.dict(exclude={"incident", "case", "creator", "artifacts"})
     )
 
-    incident = incident_service.get(db_session=db_session, incident_id=instance_in.incident.id)
-    instance.incident = incident
+    if instance_in.incident:
+        incident = incident_service.get(db_session=db_session, incident_id=instance_in.incident.id)
+        instance.incident = incident
+
+    if instance_in.case:
+        case = case_service.get(db_session=db_session, case_id=instance_in.case.id)
+        instance.case = case
 
-    workflow = get(db_session=db_session, workflow_id=instance_in.workflow.id)
     instance.workflow = workflow
 
-    creator = participant_service.get_by_incident_id_and_email(
-        db_session=db_session, incident_id=incident.id, email=instance_in.creator.individual.email
-    )
-    instance.creator = creator
+    if instance_in.creator:
+        creator = participant_service.get_by_incident_id_and_email(
+            db_session=db_session,
+            incident_id=incident.id,
+            email=instance_in.creator.individual.email,
+        )
```

Comment:
How's this going to work for cases where we don't have participants (yet)?

Reply:
We don't pass a creator for cases. This is only here to support the existing incident use case (where we do have participants). When/if we add participants to cases we can pass a creator when creating workflows from cases.